modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/SystemFDSEntryRegistryImpl.java

```diff
@@ -34,125 +29,63 @@ public class SystemFDSEntryRegistryImpl implements SystemFDSEntryRegistry {
 
 	@Override
 	public Map<String, SystemFDSEntry> getSystemFDSEntries() {
-		return _systemFDSEntries.get();
+		if (!_opened) {
```

Comment:
Not related to this PR per se but, why do you have a `getSystemFDSEntries()` that returns the whole map and also a `getSystemFDSEntry(String fdsName)` that looks up the map? It's a kind of duplication...

Reply:
I explain the reasons in the other comment